### PR TITLE
Refactor/pagination utils

### PR DESCRIPTION
**Pull request by maartensijmkens**
Refactor pagination/query utils

### CODE REVIEW

File: src/utils/typeorm/mappers/sort-direction-to-order.mapper.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { SortDirection } from '../../query/search.query.js'
+
+export function typeormOrder (direction: SortDirection): 'ASC' | 'DESC'
+export function typeormOrder (direction: undefined): undefined
+export function typeormOrder (direction: SortDirection | undefined): 'ASC' | 'DESC' | undefined
+export function typeormOrder (direction: SortDirection | undefined): 'ASC' | 'DESC' | undefined {
```
Comment (siemenvdn) [inline src/utils/typeorm/mappers/sort-direction-to-order.mapper.ts:6]: duplicate line

File: src/utils/typeorm/mappers/paginated-offset-query-to-pagination.mapper.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import type { PaginatedOffsetQuery } from '../../pagination/offset/paginated-offset.query.js'
+
+export interface TypeormPagination {
+  skip: number
+  take: number
+}
+
+export function typeormPagination (
+  query?: PaginatedOffsetQuery | null,
+  maxLimit = 25
```
Comment (siemenvdn) [inline src/utils/typeorm/mappers/paginated-offset-query-to-pagination.mapper.ts:10]: Maybe it's better to make 1 global configuration file for things of this kind

**Review by siemenvdn — APPROVED** [review]

File: src/utils/typeorm/mappers/sort-direction-to-order.mapper.ts (hunk 1 above)
Comment (maartensijmkens) [reply src/utils/typeorm/mappers/sort-direction-to-order.mapper.ts:6]: intended, this is how method overloading works

Comment (siemenvdn) [reply src/utils/typeorm/mappers/sort-direction-to-order.mapper.ts:6]: line 5 & 6 are exactly the same, u dont need to define the same signature twice

Comment (jorenvandeweyer) [reply src/utils/typeorm/mappers/sort-direction-to-order.mapper.ts:6]: you do need to define this twice if you want to expose it. @siemenvdn

**Review by jorenvandeweyer — APPROVED** [review]

File: src/utils/typeorm/mappers/sort-direction-to-order.mapper.ts (hunk 1 above)
Comment (maartensijmkens) [reply src/utils/typeorm/mappers/sort-direction-to-order.mapper.ts:6]: exactly what Joren says, we want to expose a version that accepts `SortDirection | undefined` and not only a version that accepts `SortDirection` and a version that accept `undefined`

File: src/utils/typeorm/mappers/paginated-offset-query-to-pagination.mapper.ts (hunk 2 above)
Comment (maartensijmkens) [reply src/utils/typeorm/mappers/paginated-offset-query-to-pagination.mapper.ts:10]: I agree but I'm not sure where to put it

**Review by Kobe-Kwanten — APPROVED** [review]

**Pull request merged**